db/migrate/20150107044854_users_without_location_cleanup.rb

```diff
@@ -0,0 +1,9 @@
+class UsersWithoutLocationCleanup < ActiveRecord::Migration
+  def up
+    User.delete_all(location: [nil,''])
```

Comment:
I thought of the same, so asked there in comment what should be the workaround.. Since @rohitpaulk mentioned db cleanup job, i went on with it.. Dummy value is preferred anyday..